hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -308,102 +284,87 @@
+      ]
+    },
+    { name: "COMMAND",
+      desc: '''Command Register
+
+               Parameters specific to each command segment.  One common register for all
+               devices''',
+      swaccess: "rw",
+      hwaccess: "hro",
+      hwqe: "true",
+      fields: [
+        { bits: "13:12",
+          name: "DIRECTION",
+          desc: '''The direction for the following command: "0" = Dummy cycles
+                   (no TX/RX). "1" = Rx only, "2" = Tx only, "3" Bidirectional
+                   Tx/Rx (Standard SPI mode only).'''
+          resval: "0x0"
+        }
+        { bits: "11:10",
+          name: "SPEED",
+          desc: '''The speed for this command segment: "0" = Standard SPI. "1" = Dual SPI.
+                   "2"=Quad SPI,  "3": RESERVED.''',
+          resval: "0x0"
+        },
+        { bits: "9",
+          name: "CSAAT",
+          desc: '''Chip select active after transaction.  If CSAAT = 0, the
+                   chip select line is raised immediately at the end of the
+                   command segment.   If !!COMMAND.CSAAT = 1, the chip select
+                   line is left low at the end of the current transaction
+                   segment.  This allows the creation longer, more
+                   complete SPI transactions, consisting of several separate
+                   segments for issuing instructions, pausing for dummy cycles,
+                   and transmitting or receiving data from the device.''',
+                   resval: "0x0"
+        },
+        { bits: "8:0",
+          name: "LEN",
+          desc: '''Segment Length.
+
+                   For read or write segments, this field controls the
+                   number of 1-byte bursts to transmit and or receive in
+                   this command segment.  The number of cyles required
+                   to send or received a byte will depend on !!COMMAND.SPEED.
+                   For dummy segments, (!!COMMAND.DIRECTION == 0), this register
+                   controls the number of dummy cycles to issue.
+                   ''',
+          resval: "0x0"
+        },
+      ]
     },
     { window: {
-        name: "TXDATA",
+        name: "DATA",
         items: "1",
         validbits: "32",
         byte-write: "true",
-        desc: '''SPI Transmit Data.
+        desc: '''SPI Transmit and Receive Data.
+
+                 Write data to this window to place it in the TXFIFO.
+                 Reads from this register pull data from the RXFIFO.
+                 Byte-enables are supported for writes.
 
                  The serial order of bit transmission
                  is chosen to match SPI flash devices. Individual bytes
                  are always transmitted with the most significant bit first.
                  Multi-byte writes are also supported, and if ByteOrder = 0,
-                 the bits of !!TXDATA are transmitted strictly in order of
+                 the bits of !!DATA are transmitted strictly in order of
```

Comment:
sorry i think i'm really confused by this statement. 
The first part seems to say the most significant bits of a byte are always transmitted first, but then the second part seems to say this is only the case when `ByteOrder` is set to 0?

Reply:
there's a chance i'm getting confused between byte ordering and bit ordering.
I think what you're saying is that bit ordering within a byte is always fixed, but the byte ordering itself may be swapped?

Reply:
That's right.  Please let me know if there is any way I can make the text more clear.

Reply:
I think if you just follow on the "individual bytes are always..." sentence with something like "byte order within DATA is determined ...", that would be good.

Because I think the ByteOrder = 0 case you give after is almost like an example.

Reply:
Is this any better?
```
                 The serial order of bit transmission is chosen to accomdate SPI flash devices.
                 Each individual byte is transmitted with the most significant bit first.
                 For multi-byte writes, the order of byte transmission is determined by the
                 ByteOrder parameter. If ByteOrder = 0, the most significant byte (!!DATA[31:24])
                 is transmitted first, starting with bit 31.  If ByteOrder = 1, the least
                 significant byte (!!DATA[7:0]) is transmitted first, starting with bit 7.

                 For Little-Endian processors, (such as Ibex) setting ByteOrder = 1 will ensure that
                 data is transmitted in the same order as it is arranged in processor memory, even
                 when using 32-bit instructions to copy the data.'''
```

Reply:
looks good!